java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,691 @@
+          build(newArrayTree.type());
+        }
+        build(Lists.reverse(newArrayTree.dimensions()));
+        build(Lists.reverse(newArrayTree.initializers()));
+        break;
+      default:
+        throw new UnsupportedOperationException(tree.kind().name());
+    }
+
+  }
+
+  private Block createUnconditionalJump(Tree terminator, @Nullable Block target) {
+    Block result = createBlock();
+    result.terminator = terminator;
+    if (target != null) {
+      result.successors.add(target);
+    }
+    return result;
+  }
+
+  private void buildCondition(Tree syntaxNode, Block trueBlock, Block falseBlock) {
+    switch (syntaxNode.kind()) {
+      case CONDITIONAL_OR: {
+        BinaryExpressionTree e = (BinaryExpressionTree) syntaxNode;
+        // process RHS
+        buildCondition(e.rightOperand(), trueBlock, falseBlock);
+        falseBlock = currentBlock;
+        // process LHS
+        currentBlock = createBranch(e, trueBlock, falseBlock);
+        buildCondition(e.leftOperand(), trueBlock, falseBlock);
+        break;
+      }
+      case CONDITIONAL_AND: {
+        // process RHS
+        BinaryExpressionTree e = (BinaryExpressionTree) syntaxNode;
+        buildCondition(e.rightOperand(), trueBlock, falseBlock);
+        trueBlock = currentBlock;
+        // process LHS
+        currentBlock = createBranch(e, trueBlock, falseBlock);
+        buildCondition(e.leftOperand(), trueBlock, falseBlock);
+        break;
+      }
+      // Skip syntactic sugar:
+      case PARENTHESIZED_EXPRESSION:
+        buildCondition(((ParenthesizedTree) syntaxNode).expression(), trueBlock, falseBlock);
+        break;
+      default:
+        build(syntaxNode);
+        break;
+    }
+  }
+
+  private Block createBranch(Tree terminator, Block trueBranch, Block falseBranch) {
+    Block result = createBlock();
+    result.terminator = terminator;
+    result.successors.add(trueBranch);
+    result.successors.add(falseBranch);
+    return result;
+  }
+
+  public void debugTo(PrintStream out) {
+    for (Block block : Lists.reverse(blocks)) {
+      if (block.id != 0) {
+        out.println("B" + block.id + ":");
+      } else {
+        out.println("B" + block.id + " (Exit) :");
+      }
+      int i = 0;
+      for (Tree tree : block.elements()) {
+        out.println("  " + i + ": " + syntaxNodeToDebugString(tree));
+        i++;
+      }
+      if (block.terminator != null) {
+        out.println("  T: " + syntaxNodeToDebugString(block.terminator));
+      }
+      if (!block.successors.isEmpty()) {
+        out.print("  Successors:");
+        for (Block successor : block.successors) {
+          out.print(" B" + successor.id);
+        }
+        out.println();
+      }
+    }
+    out.println();
+  }
+
+  private static String syntaxNodeToDebugString(Tree syntaxNode) {
+    StringBuilder sb = new StringBuilder(syntaxNode.kind().name())
+      .append(' ').append(Integer.toHexString(syntaxNode.hashCode()));
+    switch (syntaxNode.kind()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)